src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/jpa/repository/ChildGroupRepository.kt

```diff
@@ -1,11 +1,15 @@
 package uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository
 
-import org.springframework.data.repository.CrudRepository
-import uk.gov.justice.digital.hmpps.externalusersapi.model.ChildGroup
+import kotlinx.coroutines.flow.Flow
+import org.springframework.data.repository.kotlin.CoroutineCrudRepository
+import uk.gov.justice.digital.hmpps.externalusersapi.r2dbc.data.ChildGroup
+import java.util.UUID
 
-interface ChildGroupRepository : CrudRepository<ChildGroup, String> {
+interface ChildGroupRepository : CoroutineCrudRepository<ChildGroup, String> {
 
-  fun findByGroupCode(groupCode: String?): ChildGroup?
+  suspend fun findByGroupCode(groupCode: String?): ChildGroup?
 
-  fun deleteByGroupCode(groupCode: String?)
+  suspend fun deleteByGroupCode(groupCode: String)
+
+  fun findAllByGroup(groupId: UUID?): Flow<ChildGroup>
```

Comment:
Why do we wrap multiple values in Flow, but return actual values when there is only a single entity? This does not appear to be consistent across all repositories (i.e. UserRepository). Also repositories implemented using DatabaseClient wrap return in either Mono or Flow

Reply:
We tried to replicate how the CoroutineCrudRepository works.  I think we struggled with the databaseclient - but maybe that still need looking at

Reply:
TODO -> Check repository calls: Flow<T> should not suspend